Extract rpms

staticf0x opened the pull request:
This is filed for the intention of code review. This PR is not to be merged (although nothing bad would happen).

staticf0x committed:
RPM lockfile support

This commit add initial support for RPM lockfile
(rpms.in.yaml and rpms.lock.yaml) via lockFileMaintenance mode
by calling rpm-lockfile-prototype script.

staticf0x committed:
Add schema for rpms.lock.yaml

staticf0x committed:
Extract RPM dependencies

staticf0x committed:
Implement the RPM lockfile datasource

staticf0x committed:
Allow individual RPM updates

staticf0x committed:
Update tests

staticf0x committed:
Add recommended config to the readme

qixiang commented on lib/modules/manager/rpm/extract.ts:
```diff
@@ -1,5 +1,35 @@
+    return;
+  }
+
+  cmd.push(`rpm-lockfile-prototype ${packageFileName} --outfile ${outputName}`);
```
We're running `rpm-lockfile-prototype` twice (another one is in `artifacts.ts`. For some repositories, running the script can be resource and time consuming, do you think we can run it once in each task? After extracting the rpms from the original lockfile, we don't need the original lockfile anymore, how about just overriding the original lockfile?

staticf0x replied:
Unfortunately it's not possible, it would not be possible to find the package updates and would fall back to lockFileMaintenance only.

qixiang commented on lib/modules/manager/rpm/readme.md:
````diff
@@ -0,0 +1,41 @@
+```json
+{
+  "packageRules": [
+    {
+      "groupName": "RPM updates",
+      "matchManagers": ["rpm"],
+      "commitMessageAction": "",
+      "commitMessageTopic": "RPM updates",
+      "prTitle": "RPM updates"
+    }
````
Are we able to customize the source branch name for the PR in rpm manager? We need to include the branch name in PR's source branch.

staticf0x replied:
This was resolved in https://github.com/konflux-ci/mintmaker/pull/90#issuecomment-2459537785

staticf0x committed:
Fix linter findings

staticf0x committed:
Change snapshot to just toMatchObject

staticf0x committed:
Update README

staticf0x committed:
Use caching-rpm-lockfile-prototype

querti committed:
Create prototype for container CVE remediation

Introduce a new class ContainerVulnerabilities responsible for adding
CVE information to container PRs. It's written analogously to the
Vulnerabilities class responsible for CVE information in OSV ecosystems.
The reason why the new functionality couldn't be combined with the
existing class is that it needs to be called after the updates have been
resolved (can be seen in extract-update.ts). The CVE evaluation logic is
also different from version-based OSV ecosystems. Lastly, creating a new
file better separetes the new logic from upstream, which will make
rebasing easier.

The evaluation logic is as follows:
For each detected dependency update:
1. If the update is not a Docker type or the image is not specified via
   digest, skip
2. Gather timestamps of when the current and updated images were created
   (using Docker API)
3. Gather all vulnerabilities affecting the repo of the updated image
4. Only keep vulnerabilities that are newer than the old image and older
   than the new image (this implies that old image was affected by a
   vulnerability and the new image fixes it)
5. Create package rules based on vulnerabilities that remain
6. Package rules will apply the CVE information to the PRs

In its current state, the new module can gather relevant dependency
updates, get the 'created' timestamps, filter vulnerabilities based on
timestamps, and create packageRules and PR notes. It is NOT able to
correctly gather the vulnerabilities affecting a given repository as
this functionality is not yet implemented. For this reason, there is a
hardcoded vulnerability in the code that was used for manual testing.

Existing methods in the Docker datasource are used to get the image
config digest and the config manifest. However, small changes were made:
getConfigDigest is no longer private to be usable in a different module,
and getImageConfig was reimplemented as getImageConfigFull, since the
original method returns only a truncated config.

The full renovate test suite is verified to pass, so the new
functionality doesn't cause any regressions in the existing code.

querti committed:
Use Container CVE scanning on all dependencies with Docker datasource

Use Docker datasource instead of only allowing Docker manager
dependencies. THere are container image updates in other managers that
would otherwise be missed.

querti committed:
Make renovate use downstream osv-offline

Add @mintmaker/osv-offline to the list of dependencies and modify
container-vulnerabilities.ts to use it instead of the upstream version.

Add a new configuration option, containerVulnerabilityAlerts, which
enables OSV alerts only for containers.

staticf0x committed:
Fix rpm manager not working in subfolders

renovate[bot] committed:
chore(deps): update docker/dockerfile docker tag to v1.13.0 (#33711)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
build(deps): update dependency ignore to v7.0.2 (#33716)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
build(deps): update dependency ignore to v7.0.3 (#33717)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

TheMrMilchmann committed:
fix(docs): correct documentation for fork processing (#33712)

RahulGautamSingh committed:
fix(manager/bundler): drop strict flag (#33713)

RahulGautamSingh committed:
feat(config): allow multiple branch/pr limits (#32556)

renovate[bot] committed:
build(deps): update opentelemetry-js monorepo (#33723)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update ghcr.io/containerbase/sidecar docker tag to v13.6.0 (#33726)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update ghcr.io/containerbase/devcontainer docker tag to v13.6.0 (#33725)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
feat(deps): update ghcr.io/renovatebot/base-image docker tag to v9.34.0 (#33728)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

RahulGautamSingh committed:
fix(workers/limits): rectify faulty if condition (#33735)

renovate[bot] committed:
build(deps): update node.js to v22.13.1 (#33736)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update github/codeql-action action to v3.28.2 (#33738)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update ghcr.io/renovatebot/base-image docker tag to v9.34.1 (#33739)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update otel/opentelemetry-collector-contrib docker tag to v0.118.0 (#33746)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency eslint-plugin-jest to v28.10.1 (#33747)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency eslint-plugin-jest to v28.11.0 (#33749)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
build(deps): update dependency better-sqlite3 to v11.8.0 (#33750)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

mervyn-mccreight committed:
feat(presets): add arrow-kt monorepo (#33740)

Xzelsius committed:
feat(presets): add monorepo html-eslint (#33730)

justo-mend committed:
docs: remove platform information from Mend-hosted App config page (#33724)

fpapado committed:
test(npm): use fs test utils in extract/pnpm.spec.ts (#33756)

justo-mend committed:
docs: added list of valid namespace codes for the cacheTtlOverride config (#33754)

Co-authored-by: Rhys Arkins <rhys@arkins.net>

renovate[bot] committed:
build(deps): update dependency cronstrue to v2.53.0 (#33762)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update codecov/codecov-action action to v5.2.0 (#33763)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
build(deps): update dependency fs-extra to v11.3.0 (#33765)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
build(deps): update dependency @renovatebot/osv-offline to v1.5.12 (#33766)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update github/codeql-action action to v3.28.3 (#33767)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

zharinov committed:
feat: Switch to `Timestamp` type (#33648)

zharinov committed:
feat: Drop older `timeStamp` field support (#33734)

renovate[bot] committed:
chore(deps): update dependency @types/node to v22.10.7 (#33780)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update python:3.13 docker digest to 6ee7975 (#33781)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

GauthierPLM committed:
feat(presets): Add nestjs/config to nest monorepo (#33785)

zharinov committed:
fix(cargo): Skip "replace" range upgrade if new version still matches (#33773)

ysangkok committed:
feat(datasource/hackage): Detect deprecation status (#33778)

renovate[bot] committed:
chore(deps): update ghcr.io/containerbase/devcontainer docker tag to v13.7.0 (#33799)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update github/codeql-action action to v3.28.4 (#33798)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update ghcr.io/containerbase/sidecar docker tag to v13.7.0 (#33802)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update codecov/codecov-action action to v5.3.0 (#33805)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

HonkingGoose committed:
docs: announce upcoming disabling of encrypted secrets (#33803)

renovate[bot] committed:
chore(deps): update dependency markdownlint-cli2 to v0.17.2 (#33813)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

mathisgauthey committed:
docs(local-dev): add `pdm install` step (#33814)

renovate[bot] committed:
feat(deps): update ghcr.io/renovatebot/base-image docker tag to v9.36.2 (#33741)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

zharinov committed:
feat(http): Create package cache provider (#33742)

viceice committed:
test: refactor and ci fix for other tests (#33815)

viceice committed:
feat(docker): forward to default entrypoint (#33819)

zharinov committed:
fix(bitrise): Streamline Zod schema (#33769)

renovate[bot] committed:
build(deps): update dependency commander to v13 (#33552)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

klippx committed:
feat(preset): adding Azure and AzureAD monorepos (SDK and auth) (#33624)

RahulGautamSingh committed:
fix(config/inherited): apply secrets (#33779)

fpapado committed:
fix(npm): extract pnpm overrides with version ranges correctly (#33727)

rarkins committed:
feat: rename allowedPostUpgradeCommands/allowPostUpgradeCommandTemplating (#32657)

viceice committed:
feat(datasource/github-runners): support ubuntu arm partner images (#33824)

renovate[bot] committed:
fix(deps): update ghcr.io/renovatebot/base-image docker tag to v9.36.3 (#33827)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

RahulGautamSingh committed:
feat(config/migration): migrate package.json config (#33633)

Co-authored-by: Michael Kriese <michael.kriese@visualon.de>

renovate[bot] committed:
chore(deps): update codecov/codecov-action action to v5.3.1 (#33829)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update ghcr.io/containerbase/devcontainer docker tag to v13.7.1 (#33830)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update ghcr.io/renovatebot/base-image docker tag to v9.36.4 (#33832)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update github/codeql-action action to v3.28.5 (#33831)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update ghcr.io/containerbase/sidecar docker tag to v13.7.1 (#33833)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update ghcr.io/renovatebot/base-image docker tag to v9.36.5 (#33836)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update python:3.13 docker digest to 137ae4b (#33838)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
build(deps): update dependency better-sqlite3 to v11.8.1 (#33841)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency @swc/core to v1.10.8 (#33845)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

Alicipy committed:
feat(manager/pre-commit): parse 'ssh://git@' in pre-commit repo (#33850)

Signed-off-by: Stefan Kraus <dev@stefankraus.org>

renovate[bot] committed:
docs: update references to renovate/renovate (#33855)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): lock file maintenance (#33856)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update containerbase/internal-tools action to v3.5.21 (#33857)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency pdm to v2.22.3 (#33858)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update ghcr.io/containerbase/devcontainer docker tag to v13.7.2 (#33860)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update actions/setup-node action to v4.2.0 (#33861)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency renovatebot/github-action to v41.0.11 (#33862)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

Churro committed:
feat(gradle): add support for implicit test suite dependencies (#33775)

renovate[bot] committed:
fix(deps): update ghcr.io/containerbase/sidecar docker tag to v13.7.2 (#33865)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update ghcr.io/renovatebot/base-image docker tag to v9.36.6 (#33868)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update ghcr.io/renovatebot/base-image docker tag to v9.36.7 (#33869)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

r0bobo committed:
feat(bazel): set sourceUrl for Bazel modules (#33840)

Co-authored-by: Michael Kriese <michael.kriese@gmx.de>

maxbrunet committed:
fix(poetry): fix Google Artifacts Registry credentials lookup (#33872)

viceice committed:
fix(gitea): platform automerge only fully works on Gitea v1.24 and Forgejo v10 (#33871)

renovate[bot] committed:
fix(deps): update ghcr.io/renovatebot/base-image docker tag to v9.36.8 (#33878)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency type-fest to v4.33.0 (#33879)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

rarkins committed:
chore(label-actions): no-merge-from-main (#33875)

Co-authored-by: Michael Kriese <michael.kriese@visualon.de>

renovate[bot] committed:
feat(deps): update ghcr.io/renovatebot/base-image docker tag to v9.37.0 (#33881)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

zharinov committed:
test(gradle): Use describe blocks for versioning tests (#33882)

zharinov committed:
fix(gradle): Accept versions with trailing separators as valid (#33884)

renovate[bot] committed:
chore(deps): update github/codeql-action action to v3.28.6 (#33886)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update github/codeql-action digest to 17a820b (#33887)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
build(deps): update dependency commander to v13.1.0 (#33891)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

fpapado committed:
feat(managers/npm): support pnpm catalogs (#33376)

Co-authored-by: Sebastian Poxhofer <secustor@users.noreply.github.com>

zharinov committed:
refactor(cache): Utils to calculate soft and hard TTL (#33844)

renovate[bot] committed:
fix(deps): update ghcr.io/renovatebot/base-image docker tag to v9.37.1 (#33894)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update ghcr.io/renovatebot/base-image docker tag to v9.37.2 (#33895)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency @swc/core to v1.10.9 (#33898)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

zharinov committed:
feat(http): Return stale cache on revalidation errors (#33852)

c0d1ngm0nk3y committed:
feat(datasource): Add datasource for buildpack registry (#32721)

Co-authored-by: Nicolas Bender <nicolas.bender@sap.com>
Co-authored-by: HonkingGoose <34918129+HonkingGoose@users.noreply.github.com>
Co-authored-by: Michael Kriese <michael.kriese@visualon.de>
Co-authored-by: Pavel Busko <pavel.busko@sap.com>
Co-authored-by: Johannes Dillmann <j.dillmann@sap.com>
Co-authored-by: Michael Kriese <michael.kriese@gmx.de>

renovate[bot] committed:
fix(deps): update ghcr.io/renovatebot/base-image docker tag to v9.37.3 (#33905)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update github/codeql-action action to v3.28.7 (#33908)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

samgiz committed:
fix(http): allow overwriting default Renovate headers (#33907)

viceice committed:
fix(nuget): don't check children of `ContainerBaseImage` (#33910)

renovate[bot] committed:
build(deps): update dependency graph-data-structure to v4.3.1 (#33912)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

RahulGautamSingh committed:
fix(manager/devcontainer): allow trailing commans in jsonc files (#33913)

viceice committed:
refactor(manager): always set package name from docker image extract (#33917)

viceice committed:
fix(ansible): don't override versioning (#33918)

viceice committed:
fix(batec): don't override versioning (#33919)

zharinov committed:
feat(gradle): Single-version ranges (#33923)

renovate[bot] committed:
chore(deps): update github/codeql-action action to v3.28.8 (#33926)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
feat(deps): update ghcr.io/renovatebot/base-image docker tag to v9.38.0 (#33924)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency @types/node to v22.10.8 (#33928)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

RahulGautamSingh committed:
feat(presets/workaround): ubuntuDockerVersioning (#33714)

Co-authored-by: Rhys Arkins <rhys@arkins.net>
Co-authored-by: Michael Kriese <michael.kriese@visualon.de>

bdovaz committed:
docs(unity3d): fix broken documentation link (#33927)

jonasrutishauser committed:
feat(manager/maven): Add replacement support (#32635)

renovate[bot] committed:
chore(deps): update dependency @types/node to v22.10.9 (#33932)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

viceice committed:
fix(manager): retain `depName` when using `registryAliases` (#33911)

mpkorstanje committed:
fix(manager/custom): Support range strategy with custom managers (#33800)

Co-authored-by: Michael Kriese <michael.kriese@gmx.de>

viceice committed:
fix: only replace prefix with registry alias (#33922)

renovate[bot] committed:
fix(deps): update ghcr.io/renovatebot/base-image docker tag to v9.38.1 (#33936)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

RahulGautamSingh committed:
feat(managers/custom): generic manager for json files (#32784)

Co-authored-by: Rhys Arkins <rhys@arkins.net>
Co-authored-by: HonkingGoose <34918129+HonkingGoose@users.noreply.github.com>

MindTooth committed:
feat(presets): add tailwindcss monorepo (#33940)

reutermj committed:
feat(bazel-module): support `bazel_dep` dependencies without the `version` parameter (#33496)

Co-authored-by: Rhys Arkins <rhys@arkins.net>
Co-authored-by: Michael Kriese <michael.kriese@gmx.de>

RahulGautamSingh committed:
chore: correctly update missing labels issue (#33903)

alsyia committed:
fix(presets): match on devDependencies or dev for dev dependencies-related presets (#33909)

Co-authored-by: Rhys Arkins <rhys@arkins.net>

zharinov committed:
feat(schema): Add logging utilities for catch calls (#33950)

zharinov committed:
feat(logger): Add `withMeta` helper (#33948)

renovate[bot] committed:
fix(deps): update ghcr.io/renovatebot/base-image docker tag to v9.38.2 (#33953)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update ghcr.io/renovatebot/base-image docker tag to v9.38.3 (#33954)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency mkdocs-material to v9.6.0 (#33955)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency mkdocs-material to v9.6.1 (#33956)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

SuperSandro2000 committed:
feat(nix): rework to work with all flake inputs (#31921)

Co-authored-by: Rhys Arkins <rhys@arkins.net>
Co-authored-by: Michael Kriese <michael.kriese@gmx.de>
Co-authored-by: Mathieu Fenniak <mathieu@fenniak.net>
Co-authored-by: Mathieu Fenniak <biziqe@mathieu.fenniak.net>

ivankatliarchuk committed:
feat(versioning): aws-eks-addon versioning added (#33301)

Signed-off-by: ivan katliarchuk <ivan.katliarchuk@gmail.com>
Co-authored-by: Michael Kriese <michael.kriese@gmx.de>

philippe-granet committed:
feat(gitlab): add branch status check attemps (#32692)

Co-authored-by: Rhys Arkins <rhys@arkins.net>
Co-authored-by: Michael Kriese <michael.kriese@gmx.de>

viceice committed:
docs(versioning): fix types and tests (#33958)

zharinov committed:
feat(cloudbuild): Parse config using Zod schema (#33768)

JordanLongstaff committed:
feat(preset): Add monorepo preset for KSP (#33943)

burritobill committed:
feat(manager): add devbox manager module (#33638)

Co-authored-by: Michael Kriese <michael.kriese@gmx.de>

secustor committed:
refactor(workers): move bump version to separate function (#33965)

RahulGautamSingh committed:
refactor(process/libyear): early return if no `releaseTimestamp` for current version (#33967)

viceice committed:
refactor(schema): simplify handling (#33963)

pedromfmachado committed:
feat(presets): Added `GRADLE_OPTS` to safe global env (#33968)

Co-authored-by: Rhys Arkins <rhys@arkins.net>

viceice committed:
feat(jsonata): support yaml (#33972)

renovate[bot] committed:
build(deps): update dependency google-auth-library to v9.15.1 (#33974)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

earl-warren committed:
chore(documentation): s/Forjeo/Forgejo/ (#33976)

zharinov committed:
test(logger): Fix mock for `withMeta` function (#33979)

zharinov committed:
refactor(devbox): Refactor schema and improve logging (#33978)

Co-authored-by: Will Brennan <will.brennan@cultureamp.com>
Co-authored-by: Will Brennan <will@moondev.co.uk>
Co-authored-by: Michael Kriese <michael.kriese@gmx.de>
Co-authored-by: Michael Kriese <michael.kriese@visualon.de>

renovate[bot] committed:
build(deps): update aws-sdk-js-v3 monorepo (#33980)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

RahulGautamSingh committed:
feat: enhance libyear output (#33981)

amitdahan committed:
feat(presets): add `react-spectrum` monorepo (#33892)

rarkins committed:
feat(autoreplace): extended logging for autoreplace mismatches (#33984)

Co-authored-by: Michael Kriese <michael.kriese@visualon.de>

renovate[bot] committed:
chore(deps): update dependency google-github-actions/auth to v2.1.8 (#33987)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update containerbase/internal-tools action to v3.5.22 (#33988)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

myersg86 committed:
docs: fix typos in documentation (#33982)

Co-authored-by: greg <gmyers@gitlab.com>
Co-authored-by: Rhys Arkins <rhys@arkins.net>

RahulGautamSingh committed:
fix(manager/jsonata): populate extract result with manager config fields (#33992)

viceice committed:
build(docker): use pnpm instead of corepack (#34007)

szpak committed:
docs: language tweaks (#34005)

Co-authored-by: Michael Kriese <michael.kriese@visualon.de>

renovate[bot] committed:
chore(deps): update ghcr.io/containerbase/devcontainer docker tag to v13.7.5 (#34006)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency @swc/core to v1.10.11 (#34008)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

dsm23 committed:
feat(presets): add payloadcms monorepo (#33969)

renovate[bot] committed:
docs: update references to renovate/renovate (#33997)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

viceice committed:
chore: fix devcontainer (#34009)

rarkins committed:
chore(label-actions): improve logs, pr closing (#33995)

renovate[bot] committed:
fix(deps): update dependency mkdocs-material to v9.6.2 (#34010)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

maxbrunet committed:
feat(pip-compile): support `uv pip compile` (#33952)

Co-authored-by: Rhys Arkins <rhys@arkins.net>

mblum14 committed:
fix(manager/terragrunt): Update terragrunt packageName (#33810)

Co-authored-by: Rhys Arkins <rhys@arkins.net>

renovate[bot] committed:
chore(deps): update dependency @types/lodash to v4.17.15 (#34020)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

viceice committed:
fix(gitea): allow overriding platform version (#34019)

renovate[bot] committed:
chore(deps): update ghcr.io/containerbase/devcontainer docker tag to v13.7.6 (#34021)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update python:3.13 docker digest to d59d263 (#34026)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

risu729 committed:
feat(manager): add elixir as a mise core tooling (#34025)

Churro committed:
feat(gradle): add support for gradle repository content descriptors (#33692)

Co-authored-by: Michael Kriese <michael.kriese@visualon.de>

renovate[bot] committed:
chore(deps): update otel/opentelemetry-collector-contrib docker tag to v0.119.0 (#34032)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

viceice committed:
fix(gitea): use endpoint for pr cache pagination (#34022)

renovate[bot] committed:
chore(deps): update linters to v8.22.0 (#33880)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: secustor <sebastian@poxhofer.at>

maxbrunet committed:
docs(managers/pip-compile): update list of source files and managers (#34030)

renovate[bot] committed:
chore(deps): update sigstore/cosign-installer action to v3.8.0 (#34034)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): lock file maintenance (#33998)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

zharinov committed:
feat(npm): Support for new option `replacementApproach` (#34018)

Co-authored-by: Rhys Arkins <rhys@arkins.net>

viceice committed:
docs(gitea): fix `gitUrl` support (#34038)

renovate[bot] committed:
chore(deps): update ghcr.io/containerbase/devcontainer docker tag to v13.7.7 (#34043)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update python:3.13 docker digest to 7be9b75 (#34045)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update ghcr.io/containerbase/devcontainer docker tag to v13.7.8 (#34046)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update containerbase/internal-tools action to v3.8.0 (#33999)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
build(deps): update dependency semver to v7.7.0 (#34047)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update containerbase/internal-tools action to v3.8.2 (#34048)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update ghcr.io/renovatebot/base-image docker tag to v9.38.4 (#34049)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update ghcr.io/renovatebot/base-image docker tag to v9.38.5 (#34050)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
build(deps): update dependency @pnpm/parse-overrides to v1000.0.2 (#33996)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency @swc/core to v1.10.12 (#34053)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency renovatebot/github-action to v41.0.12 (#34001)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update ghcr.io/renovatebot/base-image docker tag to v9.38.6 (#34056)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

Flydiverny committed:
docs(jsonata): fix code block title for jsonata example (#34055)

burritobill committed:
fix(devbox): add devbox install tool to containerbase (#34002)

acaire committed:
feat(manager/asdf): add asdf conftest support (#34057)

viceice committed:
fix(datasource/dart-version): skip old svn revisions (#34059)

renovate[bot] committed:
build(deps): update dependency openpgp to v6.1.0 (#34062)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
build(deps): update dependency cronstrue to v2.54.0 (#34063)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update devcontainers/ci action to v0.3.1900000405 (#34067)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update devcontainers/ci action to v0.3.1900000408 (#34068)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update jaegertracing/jaeger docker tag to v2.3.0 (#34069)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update devcontainers/ci action to v0.3.1900000411 (#34072)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update pnpm/action-setup action to v4.1.0 (#34073)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update containerbase/internal-tools action to v3.8.4 (#34074)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency @types/diff to v7.0.1 (#34075)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update python docker tag to v3.13.2 (#34077)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update python:3.13 docker digest to 236d5c6 (#34079)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update ghcr.io/renovatebot/base-image docker tag to v9.38.7 (#34080)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update python:3.13 docker digest to 589ed66 (#34084)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update dependency mkdocs-material to v9.6.3 (#34085)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

rarkins committed:
fix(manager/custom): Revert "fix(manager/custom): Support range strategy with custom managers" (#34083)

ssams committed:
feat(manager/flux): support chartRef in HelmRelease resources (#31962)

renovate[bot] committed:
chore(deps): update github/codeql-action action to v3.28.9 (#34090)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update devcontainers/ci action to v0.3.1900000413 (#34093)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

rarkins committed:
chore(eslint): test unused-directives in CI (#34092)

renovate[bot] committed:
chore(deps): update devcontainers/ci action to v0.3.1900000416 (#34098)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update ghcr.io/containerbase/devcontainer docker tag to v13.7.9 (#34104)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update devcontainers/ci action to v0.3.1900000417 (#34103)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

rarkins committed:
chore(tests): lint-fix (#34091)

Gabriel-Ladzaretti committed:
feat(rawExec): add custom data listeners support (#34066)

Co-authored-by: Rhys Arkins <rhys@arkins.net>

renovate[bot] committed:
fix(deps): update ghcr.io/renovatebot/base-image docker tag to v9.38.8 (#34112)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

stephenwade committed:
docs: Update "noisy" schedule examples to AWS SDK v3 (#34110)

renovate[bot] committed:
chore(deps): update dependency pnpm to v9.15.5 (#34116)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
docs: update references to renovate/renovate (#34119)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): lock file maintenance (#34120)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): lock file maintenance (#34121)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency renovatebot/github-action to v41.0.13 (#34122)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

zharinov committed:
feat(cache): Relax requirements for cacheable response headers (#34134)

zharinov committed:
feat(http): Create memory cache provider (#33900)

renovate[bot] committed:
chore(deps): update linters to v8.23.0 (#34136)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
build(deps): update dependency semver to v7.7.1 (#34139)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency @swc/core to v1.10.14 (#34143)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

RahulGautamSingh committed:
docs: use `indigo` color scheme for dark mode (#34145)

secustor committed:
test: set Node timezone in jest tests (#34141)

renovate[bot] committed:
chore(deps): update containerbase/internal-tools action to v3.8.5 (#34148)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update ghcr.io/renovatebot/base-image docker tag to v9.38.9 (#34149)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update ghcr.io/renovatebot/base-image docker tag to v9.38.10 (#34150)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

nvtnlucie committed:
Merge branch 'renovatebot:main' into rpm-lockfiles-new

staticf0x committed:
Fix imports after rebase
